docs/user_guide/getting_started/minimal_script.rst

```diff
@@ -0,0 +1,115 @@
+.. currentmodule:: globus_sdk
+
+.. _minimal_script_tutorial:
+
+Creating A Minimal Script
+=========================
+
+This is a basic tutorial in the use of the Globus SDK.
+
+You can jump right in by using the ``CLIENT_ID`` seen in the example code blocks below!
+That is the ID of the tutorial client, which lets you get started quickly and easily.
+
+When you are ready to create your own application, follow :ref:`the tutorial on
+registering a Native App <tutorial_register_app>` and use its ``CLIENT_ID`` in
+the rest of the tutorial to get your own app setup.
+
+For readers who prefer to start with complete working examples, jump ahead to the
+:ref:`example scripts <minimal_script_complete_examples>` at the end before reviewing
+the doc.
+
+Define your App Object
+----------------------
+
+Accessing Globus APIs as a user requires that you login to your new app and get
+it tokens, credentials providing access the service.
+
+The SDK provides a construct which represents an application. A
+:class:`GlobusApp` is an object which can respond to requests for new or
+existing tokens and store those tokens (by default, in ``~/.globus/app/``).
+
+:class:`UserApp` is the type of :class:`GlobusApp` used for human user,
+login-driven scenarios -- this is always the right type of application object
+to use when you want to interact with services using your own account.
+
+Start by defining an application object using :class:`UserApp`:
+
+.. code-block:: python
+
+    import globus_sdk
+
+    # this is the tutorial client ID
+    # replace this string with your ID for production use
+    CLIENT_ID = "61338d24-54d5-408f-a10d-66c06b59f6d2"
+
+    # create your app
+    my_app = globus_sdk.UserApp("my-user-app", client_id=CLIENT_ID)
+
+
+.. note::
+
+    The default behavior for a :class:`UserApp` is to do a CLI-based login flow.
+    This behavior can be disabled or customized in numerous ways.
+
+    For the full menu of options, look at the documentation about :ref:`Using a
+    GlobusApp! <using_globus_app>`
+
+Access the APIs via Clients
+---------------------------
+
+Once you have an app defined, you can use it with client objects to access
+various Globus APIs.
+
+When you attempt to interact with a service, the app will automatically prompt
+you to login.
+
+Start by defining the client object:
+
+.. code-block:: python
+
+    groups_client = globus_sdk.GroupsClient(app=my_app)
+
+And now you can use it to perform some simple interaction, like listing
+your groups:
+
+.. code-block:: python
+
+    # print in CSV format
+    # ('name' could actually have commas in it, so it is quoted)
+    print("ID,Name,Roles")
+    for group in groups_client.get_my_groups():
+        roles = "|".join({m["role"] for m in group["my_memberships"]})
+        print(",".join([group["id"], f'"{group["name"]}"', roles]))
+
+When ``groups_client.get_my_groups()`` runs in the example above, the SDK
+will prompt you to login.
```

Comment:
```suggestion
When ``groups_client.get_my_groups()`` runs in the example above, the SDK
prompts the user to login.
```

Reply:
Per above, I disagree with this change. The target user is the reader of the doc. Simplifying the model to "you write the code, you login" helps keep the scope tighter.

"You write the code, they login" can be probably be served by identical code, but it's a different target use case.